### ISSUE TITLE
pyright cannot find dependencies, requires `--pythonpath`

### ISSUE THREAD

**Issue by jorenham** [hauntsaninja/mypy_primer#141]
When I locally run 

```shell
uv run mypy_primer --type-checker pyright -k scipy-stubs --debug --clear
```

around ~37068 errors are reported. It's only the first two errors that are relevant:

```
/tmp/mypy_primer/projects/scipy-stubs/scipy-stubs/__init__.pyi
  /tmp/mypy_primer/projects/scipy-stubs/scipy-stubs/__init__.pyi:4:6 - error: Import "numpy" could not be resolved (reportMissingImports)
/tmp/mypy_primer/projects/scipy-stubs/scipy-stubs/_lib/_disjoint_set.pyi
  /tmp/mypy_primer/projects/scipy-stubs/scipy-stubs/_lib/_disjoint_set.pyi:5:8 - error: Import "optype" could not be resolved (reportMissingImports)
```

So it looks like pyright isn't able to find the (installed) `numpy` and `optype` dependencies.

It's easy to confirm this by changing https://github.com/hauntsaninja/mypy_primer/blob/ba3a8ed842213b8bc877ce9f5a21859f96616f4d/mypy_primer/projects.py#L806 to

```py
pyright_cmd="{pyright} --pythonpath=/tmp/mypy_primer/projects/_scipy-stubs_venv/bin/python",
```

If we runn the same `mypy_primer` command again, we'll now see 43 errors (caused by the missing `packaging` dependency, I'll submit a PR soon).

---

I'm guessing that `scipy-stubs` isn't the only project with this issue. I'm not sure what the best solution here would be, but hardcoding the `--pythonpath` probably isn't the way to go here.

**Comment by jorenham** [comment]
What might also be relevant is the following output

```
old
> source /tmp/mypy_primer/projects/_scipy-stubs_venv/bin/activate; /tmp/mypy_primer/old_pyright/pyright/packages/pyright/index.js --pythonpath=/tmp/mypy_primer/projects/_scipy-stubs_venv/bin/python (5.8s)
        /bin/sh: 1: source: not found
        0 errors, 0 warnings, 0 informations 
```

Specifically, the `source: not found`, which could be an indication that the virtualenv isn't activated when running `pyright`.

**Comment by hauntsaninja** [comment]
Hmm it works fine for me, can you check if https://github.com/hauntsaninja/mypy_primer/pull/140 resolves for you?

**Comment by jorenham** [comment]
I don't have much time today, but I'll try to check it tonight (in 9 hours or so)

**Comment by jorenham** [comment]
Yes, that indeed seems to solve it 🎉
